### ISSUE TITLE
`override` is ignored in `as_pkgdown()` on `pkgdown` objects

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/pkgdown#2257]
## Reprex

``` r
td <- tempfile("testpkg")

usethis::create_package(td)
#> ✔ Creating '/tmp/RtmpgHN77D/testpkg6642792d7bf9/'
#> ✔ Setting active project to '/tmp/RtmpgHN77D/testpkg6642792d7bf9'
#> ✔ Creating 'R/'
#> ✔ Writing 'DESCRIPTION'
#> Package: testpkg6642792d7bf9
#> Title: What the Package Does (One Line, Title Case)
#> Version: 0.0.0.9000
#> Authors@R (parsed):
#>     * First Last <first.last@example.com> [aut, cre] (YOUR-ORCID-ID)
#> Description: What the package does (one paragraph).
#> License: `use_mit_license()`, `use_gpl3_license()` or friends to pick a
#>     license
#> Encoding: UTF-8
#> Roxygen: list(markdown = TRUE)
#> RoxygenNote: 7.2.3
#> ✔ Writing 'NAMESPACE'
#> ✔ Setting active project to '<no active project>'

p <- pkgdown::as_pkgdown(td)

p_es <- pkgdown::as_pkgdown(p, override = list(lang = "es"))

identical(p, p_es)
#> [1] TRUE

p_es$meta$lang
#> NULL
```

<sup>Created on 2022-12-19 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

## Use case

I'm trying to override parameters in `build_site_github_pages()`. But the `pkg` argument is converted to a `pkgdown` object before passing it to `build_site()`, where the `override` argument should be used. As a result, the `override` argument is silently ignored.

Using `build_site()` directly may be a better choice for this use case but it's still confusing that an argument gets ignored without warning or message.

**Comment by dshemetov** [comment]
I don't think this is fully fixed in #2406. `as_pkgdown(pkg, override)`, if `pkg = "."`, computes a lot of top level attribute variables that are used downstream that don't live in `pkg$meta`, such as `destination` and `dst_path`. Before #2406, if `is_pkg(pkg)` was true, `as_pkgdown(pkg, override)` would just return `pkg`. With #2406, we modify `pkg$meta <- modify_list(pkg$meta, override)`, but this doesn't recompute the downstream attributes. This leads to an instance where version_label does not get properly overwritten with this code with `pkgdown==2.0.9`

```r
# _pkgdown.yml
development:
  mode: devel
  version_label: success

# Code
r$> Sys.setenv("PKGDOWN_DEV_MODE" = "release")
r$> override = list(development = list(mode = "release", version_label = "light"))
r$> pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE, override=override)
```

So a bit more work needs to be done to go through the rest of the `as_pkgdown()` code and tease out which attributes need to be recomputed if `meta` changes.

**Comment by dshemetov** [comment]
For a less general solution that targets only `build_site_github_pages()`, we could merge an `override` arguments in the dots with `list(destination = dest_dir)` and feed that to the initial call of `as_pkgdown()`. See [here](https://github.com/r-lib/pkgdown/blob/2dc1877830c823bed1aa89546e84fbd95141a818/R/build-github.R#L19-L25).
